server/etcdserver/server.go

```diff
@@ -971,6 +972,7 @@ func (s *EtcdServer) applySnapshot(ep *etcdProgress, apply *apply) {
 	// Eventually the new consistent_index value coming from snapshot is overwritten
 	// by the old value.
 	s.consistIndex.SetBackend(newbe)
+	s.verifySnapshot(apply.snapshot)
```

Comment:
This is pretty specific verification in the context of recently populated s.consistencyIndex from the snapshot. 
It's generic truth that last loaded snapshot should has `index < cindex`. 

Please consider: 

```
   verifySnapshotIndex(apply.snapshot, s.consistIndex.ConsistentIndex())
```

```
func verifySnapshotIndex(snapshot raftpb.Snapshot, cindex int) {
	verify.Verify(func() {
		if cindex != snapshot.Metadata.Index { ... }
    }
}
```